### PR TITLE
 [XHarness] Reneable the SystemNetHttpTests on iOS, TvOS and WatchOS.

### PR DESCRIPTION
**Pull request by mandel-macaque**
There are two main reasons why the tests are ginored:

1. The tests do not use https.
2. WatchOS is running the iOS tests until we have the tests in the SDK
for the watch runtime.

Fixes https://github.com/xamarin/maccore/issues/1144
Fixes https://github.com/xamarin/maccore/issues/1145

### CODE REVIEW

**Comment by mandel-macaque** [comment]
Mono issue: https://github.com/mono/mono/issues/12430

File: tests/bcl-test/BCLTests/common-SystemNetHttpTests.ignore [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# #1
+#  Expected: instance of <System.Net.Http.HttpRequestException>
+#  But was:  <System.Net.WebException>
+MonoTests.System.Net.Http.HttpClientTest.WildcardConnect
+MonoTests.System.Net.Http.HttpClientTest.Send_Transfer_Encoding_Custom
+
+#  Error Domain=NSURLErrorDomain Code=-1022 "The resource could not be loaded because the App Transport Security policy requires the use of a secure connection."
```
Comment (rolfbjarne) [inline tests/bcl-test/BCLTests/common-SystemNetHttpTests.ignore]: You can fix this by adding this to the Info.plist for the app:

```xml
<key>NSAppTransportSecurity</key>
<dict>
    <key>NSAllowsArbitraryLoads</key>
     <true/>
</dict>
```

Comment (mandel-macaque) [reply tests/bcl-test/BCLTests/common-SystemNetHttpTests.ignore]: Already dealt with :)

**Comment by mandel-macaque** [comment]
Mono issue was added: https://github.com/mono/mono/issues/12430

**Pull request merged**